src/syft_rds/syft_runtime/main.py

```diff
@@ -303,97 +489,52 @@ def build_docker_command(self, config: JobConfig) -> list[str]:
             "--ulimit",
             "fsize=10000000:10000000",  # ~10MB file size limit
         ]
-        interpreter = " ".join(config.runtime.cmd)
-        interpreter_str = f'"{interpreter}"' if " " in interpreter else interpreter
-        return [
+
+        docker_run_cmd = [
             "docker",
             "run",
             "--rm",  # Remove container after completion
             *limits,
             # Environment variables
             "-e",
-            f"TIMEOUT={config.timeout}",
+            f"TIMEOUT={job_config.timeout}",
             "-e",
-            f"DATA_DIR={config.data_mount_dir}",
+            f"DATA_DIR={job_config.data_mount_dir}",
             "-e",
             f"OUTPUT_DIR={DEFAULT_OUTPUT_DIR}",
             "-e",
             f"INTERPRETER={interpreter_str}",
             "-e",
-            f"INPUT_FILE='{config.function_folder / config.args[0]}'",
-            *config.get_extra_env_as_docker_args(),
+            f"INPUT_FILE='{DEFAULT_WORKDIR}/code/{job_config.args[0]}'",
+            *job_config.get_extra_env_as_docker_args(),
             *docker_mounts,
-            "syft_python_runtime",
-            *config.args,
+            "--workdir",
+            DEFAULT_WORKDIR,
+            image_name,
+            f"{DEFAULT_WORKDIR}/code/{job_config.args[0]}",
+            *job_config.args[1:],
         ]
+        logger.debug(f"Docker run command: {docker_run_cmd}")
+        return docker_run_cmd
 
-    def validate_docker(self, config: JobConfig) -> bool:
-        """Validate Docker image availability"""
-        if not config.use_docker:
-            return True  # Skip Docker validation when not using Docker
 
-        image_name = "syft_python_runtime"
-        # Check Docker daemon availability
-        subprocess.run(["docker", "info"], check=True, capture_output=True)
+class SyftRunner:
```

Comment:
Earlier comment, I think this should be either called SyftRunnerFactory, or if we dont wan't to go full OOP `get_runner_cls(config) -> DockerRunner | PythonRunner`